### DISCUSSION TITLE
Allow to customize authorization url asynchronously

### DISCUSSION THREAD

**Discussion by liho00** [nextauthjs/next-auth#6902] [Ideas]
### Description 📓

I am requesting a feature to allow fetch the authoization url from backend, and set to the provider before navigating to the provider auth page.

Why need this feature? Because backend able to generate the authorization with some random key in the state query and save it in the redis cache as session. The purpose of this is to prevent the user spam the oauth url, and increase the security measure.

### How to reproduce ☕️

Current Behavior
 authorization: {
        url: "https://accounts.google.com/o/oauth2/v2/auth",
        params: {
          scope: "openid email profile",
          response_type: "code",
          redirect_uri: "http://localhost:3000/api/auth/callback/google",
        },
      },

Expected Behavior
 authorization: {
        request: () =>{
          const authUrl = API.fetch("https://accounts.google.com/o/oauth2/v2/auth" )
          return authUrl 
        }
      },

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
NextAuth.js already includes the necessary checks (See Google for example https://github.com/nextauthjs/next-auth/blob/7327468697b8821cae2ad2e04c1345ac32bb2043/packages/next-auth/src/providers/google.ts#L31)
